### ISSUE TITLE
Does evaluation on multi GPU work correctly? 

### ISSUE THREAD

**Issue by sadransh** [google/automl#768]
I am using the code on train_and_eval with these settings on 4 gpus :
 
`--mode=train_and_eval --train_batch_size=4  --eval_batch_size=4 --eval_samples=2111  num_epochs=100       --save_checkpoints_steps=3000     --eval_after_training=True --run_epoch_in_child_process=True     --strategy=gpus`

during validation I added a few lines to count number of boxes on   `self.dataset['annotations']` it is about 1/4 of the of all my annotations. So, I think it is not working correctly and results of CoCoeval when using multi gpu is on ~  1/num_of_gpus .

**Comment by fsx950223** [comment]
Eval mode doesn't have any strategy. It executes on the host.
It's your problem here.

**Comment by sadransh** [comment]
@fsx950223 so you mean when using `train_and_eval` with  ` --eval_batch_size=4` it evaluates results on whole validation set. Correct? or I should you `--eval_batch_size=1` ?

**Comment by fsx950223** [comment]
> @fsx950223 so you mean when using `train_and_eval` with ` --eval_batch_size=4` it evaluates results on whole validation set. Correct? or I should you `--eval_batch_size=1` ?

Eval has the same batch size as train mode. It's very easy to get these info from source code. Maybe you could read source code by yourself?
